clickhouse_connect/driver/httpclient.py

```diff
@@ -127,8 +131,22 @@ def exec_query(self, query: str, settings: Optional[Dict] = None, use_none: bool
         params['database'] = self.database
         if settings:
             params.update(settings)
-        response = self._raw_request(self._format_query(query), params=params, headers=headers)
-        data_result = self.parse_response(response.content, use_none)
+        columns_only = columns_only_re.search(query)
+        if columns_only:
+            final_query = query + ' FORMAT JSON'
+        else:
+            final_query = self._format_query(query)
+        response = self._raw_request(final_query, params=params, headers=headers)
+        if columns_only:
```

Comment:
nit: I would combine the 2 if cases it will lead to duplicate 

response = self._raw_request(final_query, params=params, headers=headers)

but overall will be shorter and more readable IMO